Store: add failing test for potential dedup issue

MichaHoffmann opened the pull request:
This feels like a dedup issue in the store if we have multiple blocks that contain inner replica labels. The added test fails with
```
go test . -run TestBucketStoreDedupOnBlockSeriesSet
--- FAIL: TestBucketStoreDedupOnBlockSeriesSet (0.45s)
    testutil.go:91: bucket_test.go:3430: ""

        	exp: 2

        	got: 6

FAIL
FAIL	github.com/thanos-io/thanos/pkg/store	0.492s
FAIL
```

saswatamcode reviewed (approved):
Let's iterate on a fix

fpetkovski commented:
Any way we can add this to existing tests?

MichaHoffmann commented:
> Any way we can add this to existing tests?

I think there could be a way, i added this pretty ad-hoc with a bit of copy paste to investigate if the issue is real. Can we clean it up in a followup after we fixed the issue?

saswatamcode reviewed (approved):
Let's iterate on adding this to others after we fix!
